py4j-python/src/py4j/clientserver.py

```diff
@@ -140,7 +140,7 @@ def __init__(
             ssl_context=None, auto_gc=False,
             accept_timeout=DEFAULT_ACCEPT_TIMEOUT_PLACEHOLDER,
             read_timeout=None, propagate_java_exceptions=False,
-            auth_token=None):
+            auth_token=None, auto_gc_connections=False):
```

Comment:
I checked other tests pass when `auto_gc_connections` is `True` except a couple of tests in `py4j.tests.memory_leak_test:ClientServerTest` which assumes several `ClientServerConnection`s are not garbage-collected after shutting down the client server.